essentials/src/test/java/org/dmfs/jems/single/elementary/FrozenTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.single.elementary;
+
+import org.dmfs.jems.single.Single;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertThat;
+
+
+/**
+ * Unit test for {@link Frozen}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class FrozenTest
+{
+    @Test
+    public void test()
+    {
+        Object firstValue = new Object();
+
+        Single<Object> frozen = new Frozen(new ChangingValueSingle(firstValue));
+
+        assertThat(frozen.value(), sameInstance(firstValue));
+        assertThat(frozen.value(), sameInstance(firstValue));
+        assertThat(frozen.value(), sameInstance(firstValue));
+    }
+
+
+    private static class ChangingValueSingle implements Single<Object>
+    {
+        private final Object mFirstValue;
+
+        private boolean mFirst = true;
+
+
+        private ChangingValueSingle(Object firstValue)
+        {
+            mFirstValue = firstValue;
+        }
+
+
+        @Override
+        public Object value()
+        {
+            if (mFirst)
+            {
+                mFirst = false;
+                return mFirstValue;
+            }
+            return new Object();
```

Comment:
The test does not quite test what it's supposed to. In this case it's an explicit requirement that the delegate is called only once. So strictly spoken it's not sufficient to just check that the result doesn't change. In theory it still could call the delegate (and discard the result of it).

Reply:
Updated as suggested so that mock Single throws for subsequent calls. Ready for review again.